# HG changeset patch
# User Thorsten Meyer <thorsten.meyier@gmx.de>
# Date 1224187270 -7200
#      Thu Oct 16 22:01:10 2008 +0200
# Node ID d95c4e31bb2dd050ad1f5e5e202bdccc966d1ecf
# Parent  fa78cb8d8a5c4e418384e0bce3ec90de50ca18a4
recommend limited line width in code examples

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -50,16 +50,20 @@ 2008-10-15  David Bateman  <dbateman@fre
 	* interpreter/Makefile.in (SUB_SOURCES): Add it here.
 	(POLYNOMIAL_FILES_NODIR): New varaible for polynomial example class
 	methods.
 	(EXAMPLE_FILES_NODIR): Add POLYNOMIAL_FILES_NODIR here.
 	* interpreter/octave.texi: Reference new OOP chapter here.
 	* interpreter/func.txi: Document private directories and function
 	precedence here.
 	
+2008-10-16  Thorsten Meyer  <thorsten.meyier@gmx.de>
+        * interpreter/contrib.txi: added recommendation of limited
+          line width in code examples
+          
 2008-10-03  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
 	* interpreter/contrib.txi: added example for mercurial queues
 
 08-10-03  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
         * interpreter/contrib.txi: correction of the mercurial example
         * interpreter/container.txi: minor correction of the text
diff --git a/doc/interpreter/contrib.txi b/doc/interpreter/contrib.txi
--- a/doc/interpreter/contrib.txi
+++ b/doc/interpreter/contrib.txi
@@ -301,10 +301,22 @@ For C you should obviously follow all C+
 
 If you happen to modify a Fortran file, you should stay within Fortran 77
 with common extensions like @code{END DO}. Currently, we want all sources
 to be compilable with the f2c and g77 compilers, without special flags if
 possible. This usually means that non-legacy compilers also accept the sources.
 
 The M4 macro language is mainly used for autoconf configuration files. You should
 follow normal M4 rules when contributing to these files. Some M4 files come
-from external source, namely the Autoconf archive @url{http://autoconf-archive.cryp.to}.
+from external source, namely the Autoconf archive
+@url{http://autoconf-archive.cryp.to}.
 
+If you give a code example in the documentation written in texinfo with the 
+@code{@@example} environment, you should be aware that the text within such an 
+environment will not be wrapped. It is recommended that you keep the lines
+short enough to fit on pages in the generated pdf or ps documents. Here is a 
+ruler (in an @code{@@example} environment) for finding the appropriate line 
+width:
+
+@example
+         1         2         3         4         5         6         7         8
+12345678901234567890123456789012345678901234567890123456789012345678901234567890
+@end example
